File: go/models/movie.go
package models

type Movie struct {
	Id          int    `json:"id"`
	Name        string `json:"name"`
	Genre       string `json:"genre"`
	Plot        string `json:"plot"`
	Language    string `json:"language"`
	Country     string `json:"country"`
	Rating      string `json:"rating"`
	ReleaseDate string `json:"release_date"`
	Runtime     int    `json:"runtime"`
	Poster      string `json:"string"`
	CreatedAt   string `json:"created_at"`
	UpdatedAt   string `json:"updated_at"`
}

File: php/app/Models/Movie.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\Relations\HasManyThrough;

/**
 * @property int id
 * @property string name
 * @property string genre
 * @property string plot
 * @property string language
 * @property string country
 * @property string rating
 * @property DateTimeInterface release_date
 * @property int runtime
 * @property string poster
 * @property DateTimeInterface created_at
 * @property DateTimeInterface updated_at
 */
class Movie extends Model
{
    public function theaters(): BelongsToMany
    {
        return $this->belongsToMany(Theater::class);
    }

    public function transactions(): HasManyThrough
    {
        return $this->hasManyThrough(TheaterMovieTransaction::class, TheaterMovie::class);
    }
}
